sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusAsyncConsumer.java

```diff
@@ -26,17 +30,28 @@
     private final String linkName;
     private final ServiceBusReceiveLinkProcessor linkProcessor;
     private final MessageSerializer messageSerializer;
-    private final Flux<ServiceBusReceivedMessage> processor;
+    private final Flux<ServiceBusReceivedMessage> messageSource;
 
     ServiceBusAsyncConsumer(String linkName, ServiceBusReceiveLinkProcessor linkProcessor,
-        MessageSerializer messageSerializer, int prefetch) {
+        MessageSerializer messageSerializer, int prefetch, boolean autoLockRenewal,
+        Duration maxAutoLockRenewDuration, LockContainer<LockRenewalOperation> messageLockContainer,
+        Function<String, Mono<OffsetDateTime>> onRenewLock) {
         this.linkName = linkName;
         this.linkProcessor = linkProcessor;
         this.messageSerializer = messageSerializer;
-        this.processor = linkProcessor
+
+        Flux<ServiceBusReceivedMessage> source = linkProcessor
             .map(message -> this.messageSerializer.deserialize(message, ServiceBusReceivedMessage.class))
             .publish(prefetch)
             .autoConnect(1);
+
+        if (autoLockRenewal) {
+            this.messageSource = new ServiceBusMessageRenewOperator(source,
```

Comment:
Since this operator is created only when `autoLockRenewal` is true, do we need the 2nd param?